### PR TITLE
Support docker and rkt  custom containers and custom fleet unit files

### PR DESCRIPTION
**Pull request by hectorj2f**
This PR aims to:
- [x] Allow the definition of rkt or docker applications using the benchmark YAML file
- [x] Add support to deploy custom docker containers.
- [x] Add support to deploy custom rkt containers.
- [x] Rename `BeaconObserver` to `UnitObserver`
- [x] Add support for a custom unit file service
- [x] Enable to specify: volumes, network type, image, name, args, environment vars, ports for rkt and docker containers

TODO: Add documentation.

```
application:
 name: helloworld
 image: giantswarm/helloworld
 type: docker
 network: host
 volumes:
  - source: /tmp
    target: /home
 ports:
   - 8000
 envs:
  x1: value1
  x2: value2
 args:
   - sh
   - "-c"
   - "echo \"Hello world\" > index.html && exec python3 -m http.server"
instancegroup-size: 1
instructions:
  - sleep: 1
  - start:
     max: 1
     interval: 100
  - sleep: 200
  - stop: stop-all

```

ping @giantswarm/team-core

### CODE REVIEW

**Comment by hectorj2f** [comment]
Related to: https://github.com/giantswarm/nomi/issues/10

**Comment by hectorj2f** [comment]
ping @denderello @zeisss I'd like to know your impressions about adding this feature to support custom docker and rkt containers. Do you see something missing any improvements?

File: examples/benchmarkDefDocker.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+application:
+ name: helloworld
+ image: giantswarm/helloworld
+ type: docker
+ ports:
+   - 8000
+ args:
+   - sh
+   - "-c"
+   - "echo \"Hello world\" \u003e index.html \u0026\u0026 exec python3 -m http.server"
```
Comment (zeisss) [inline examples/benchmarkDefDocker.yaml]: why can't we write plain characters here? `>` etc?

Comment (hectorj2f) [inline examples/benchmarkDefDocker.yaml]: Yes, we can, just a copy & paste of the real example

File: cmd/run.go [hunk 2]
```diff
@@ -121,24 +119,30 @@ func runRun(cmd *cobra.Command, args []string) {
...
-		return fleetPool.StartUnitGroup(unit.MakeUnitChain(id, runFlags.listenAddr, unitEngine.InstanceGroupSize(), runFlags.useDocker, runFlags.useRkt))
+		return fleetPool.StartUnitGroup(builder.MakeUnitChain(id))
 	}
 
 	unitEngine.StopFunc = func(id string) error {
 		if runFlags.verbose {
 			log.Logger().Infof("stopping unit with id %s\n", id)
 		}
-		return fleetPool.Stop(beaconUnitPrefix + "-0@" + id + ".service")
+		return fleetPool.Stop(builder.GetUnitPrefix() + "-0@" + id + ".service")
```
Comment (denderello) [inline cmd/run.go:154]: love it!

File: unit/builder.go [hunk 3]
```diff
@@ -166,14 +250,16 @@ func buildRktService(listenAddr string) []*schema.UnitOption {
 			Value:   "/usr/bin/rkt rm --uuid-file=/run/rkt-uuids/%p-%i",
 		},
 	}
+
+	return unit
 }
 
-func buildDefaultService(listenAddr string) []*schema.UnitOption {
+func (b *Builder) buildDefaultService() []*schema.UnitOption {
```
Comment (denderello) [inline unit/builder.go]: How about renaming this to `buildShellService`? I like to be explicit about it.

**Comment by denderello** [comment]
Cannot see anything missing atm. I like being able to use `rkt` now! :)

File: unit/builder.go (hunk 3 above)
Comment (hectorj2f) [inline unit/builder.go]: Totally

**Comment by hectorj2f** [comment]
Partially related to https://github.com/giantswarm/nomi/issues/14

**Comment by hectorj2f** [comment]
@zeisss @denderello I added two new features to use custom fleet unit files(one of type service), and better support for volumes for docker and rocket. I'll add tomorrow some Documentation if you give me the LGTM for the code.

**Comment by zeisss** [comment]
The code looks fine, but since I haven't seen Nomi before I have no clue if this breaks anything or matches the intended design :(

**Comment by hectorj2f** [comment]
thx @zeisss. It doesn't break the current design

**Comment by puja108** [comment]
language LGTM ;)

**Pull request merged**